src/main/java/tak/GameUpdateBroadcaster.java

```diff
@@ -0,0 +1,128 @@
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package tak;
+
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpClient.Redirect;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.concurrent.BlockingDeque;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.Flow.Subscriber;
+import java.util.concurrent.Flow.Subscription;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import tak.FlowMessages.GameUpdate;
+
+/**
+ * This class should be subscribed to all games and send out updates to HTTP endpoints
+ * of subscribers.
+ */
+public final class GameUpdateBroadcaster implements Runnable, Subscriber<GameUpdate> {
+	public static URL eventSubscriberUrl;
+
+	private Logger logger = Logger.getLogger(TakServer.class.getName());
+	private BlockingDeque<GameUpdate> updatesToBroadcast = new LinkedBlockingDeque<>();
+
+	private AtomicBoolean stopped = new AtomicBoolean(false);
+
+	public void stop() {
+		logger.info("stopping " + GameUpdateBroadcaster.class.getSimpleName() + " ...");
+		stopped.set(true);
+	}
+
+	public void run() {
+		if (eventSubscriberUrl == null) {
+			logger.severe("failed to start " + GameUpdateBroadcaster.class.getSimpleName() + " because no event-subscriber-url is defined");
+			return;
+		}
+
+		logger.info("started " + GameUpdateBroadcaster.class.getSimpleName());
+		try {
+			while(!stopped.get()) {
+				var update = updatesToBroadcast.poll(1, TimeUnit.SECONDS);
+				if (update == null) {
+					continue;
+				}
+
+				if (logger.isLoggable(Level.FINE)) {
+					logger.fine("broadcasting " + update.toString());
+				} else {
+					logger.info("broadcasting " + update.type + " game#" + update.game.id);
+				}
+
+				try {
+					var jsonMapper = new ObjectMapper()
+						.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false)
+						.setVisibility(PropertyAccessor.FIELD, Visibility.DEFAULT);
+					var jsonString = jsonMapper.writeValueAsString(update);
+					logger.fine(jsonString);
+
+					var client = HttpClient.newBuilder()
+						.connectTimeout(Duration.ofSeconds(5))
+						.followRedirects(Redirect.NORMAL)
+						.build();
+					var request = HttpRequest.newBuilder(eventSubscriberUrl.toURI())
+						.POST(BodyPublishers.ofString(jsonString))
+						.timeout(Duration.ofSeconds(15))
```

Comment:
Not sure what kind of timeout is sensible here. Maybe it doesn't matter because we don't need to wait for the reply. We don't want it block the queue for long.